### PR TITLE
VFB-194 VFB-195: Add audit logs for adding and editing packing slots and fix RLS policies checking for admin access

### PR DESCRIPTION
**Pull request by rickaw**
## What's changed
- Add audit logs for adding and editing packing slots
- Fix RLS policies checking for admin access
- Add an e2e test for moving packing slots

## Checklist
- [x] The ticket is up-to-date - Please document any deviations from the original approach if there is any.
- [x] I have documented the testing steps for QA
- [x] I have self-reviewed this PR
- [x] Make sure you've verified it works via `npm run dev`
- [x] Make sure you've verified it works via `npm run build` and `npm run start`
- [x] Make sure you've fixed all linting problems with `npm run lint_fix`
- [x] Make sure you've tested via `npm run test:e2e`

If you have made any changes to the database...
  - [x] The migration files are up-to-date with my final set up (`npx supabase db diff -f <name_of_migration>` should create nothing at this point)
  - [x] I have updated the typescript definitions for the database with `db:local:generate_types`
  - [x] I have modified the seed in `supabase/seed/seed.mts` if appropriate
  - [x] If I have modified the seed, I have also generated the seed with `npm run db:generate_seed` 
  - [x] With my final set up, I can run `npm run dev:reset_supabase` without any errors.
  - Does this require resetting the deployed database? - YES / NO (remove as appropriate)
  - [x] I have checked that migration can happen successfully without resetting the database, doing the following.
    - Make sure you have rebased your branch onto dev or merged dev into your branch.
    - Checkout the dev branch
    - Run `npm run post_checkout` to reset the database, including the seed data.
    - Checkout your branch
    - Run `npx supabase migrations up --include-all --local` to run all outstanding migration files
    - Check that the resulting database is what you expect, bar any seed data changes.

### CODE REVIEW

File: docs/e2e-testing.md
Comment (rickaw) [inline docs/e2e-testing.md:1]: We can add to this file as we write more e2e tests

File: cypress/e2e/admin/packingSlots.cy.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
+const modifyPackingSlotsText = "modify packing slots";
+
+describe("Packing slots on admins page", () => {
+    it("Move the first packing slot down and then up", () => {
+        cy.login();
+
+        cy.visit("/admin");
```
Comment (rickaw) [inline cypress/e2e/admin/packingSlots.cy.ts:7]: use before each

File: src/app/admin/packingSlotsTable/PackingSlotActions.ts [hunk 2]
```diff
@@ -42,36 +50,59 @@ const formatNewRowToDBPackingSlot = (newRow: PackingSlotRow): NewDbPackingSlot =
...
+export const insertNewPackingSlot = async (
+    newRow: PackingSlotRow
+): Promise<PackingSlotActionResult> => {
     const data = formatNewRowToDBPackingSlot(newRow);
     const { error } = await supabase.from("packing_slots").insert(data);
 
     if (error) {
-        const logId = await logErrorReturnLogId("Error with insert: Packing slots", error);
-        throw new DatabaseError("insert", "packing slots", logId);
+        const logId = await logErrorReturnLogId("Error with insert: Packing slots", {
```
Comment (rickaw) [inline src/app/admin/packingSlotsTable/PackingSlotActions.ts:60]: update message

File: src/app/admin/packingSlotsTable/PackingSlotActions.ts [hunk 3]
```diff
@@ -42,36 +50,59 @@ const formatNewRowToDBPackingSlot = (newRow: PackingSlotRow): NewDbPackingSlot =
...
         id1: row1.id,
         id2: row2.id,
     });
+
     if (error) {
-        const logId = await logErrorReturnLogId("Error with update: packing slots order", error);
-        throw new DatabaseError("update", "packing slots order", logId);
+        const logId = await logErrorReturnLogId("Failed to update packing slot order", {
+            error,
+        });
```
Comment (rickaw) [inline src/app/admin/packingSlotsTable/PackingSlotActions.ts:103]: add id1 and id2

File: src/app/admin/packingSlotsTable/PackingSlotsTable.tsx [hunk 4]
```diff
@@ -71,6 +72,20 @@ function EditToolbar(props: EditToolbarProps): React.JSX.Element {
     );
 }
 
+function getBaseAuditLogForPackingSlotSwap(
+    packingSlotRow: PackingSlotRow
+): Pick<AuditLog, "action" | "content" | "packingSlotId"> {
+    return {
+        action: "Move a packing slot down",
```
Comment (rickaw) [inline src/app/admin/packingSlotsTable/PackingSlotsTable.tsx:79]: this should be configurable

File: src/app/admin/packingSlotsTable/PackingSlotsTable.tsx [hunk 5]
```diff
@@ -129,24 +144,29 @@ const PackingSlotsTable: React.FC = () => {
...
-                    setErrorMessage("Error updating data, please try again");
-                })
-                .finally(() => setIsLoading(false));
+            const { error: updatePackingSlotError } = await updateDbPackingSlot(newRow);
+
+            if (updatePackingSlotError) {
+                setErrorMessage(
+                    `Failed to update the packing slot. Log ID: ${updatePackingSlotError.logId}`
+                );
+            }
```
Comment (rickaw) [inline src/app/admin/packingSlotsTable/PackingSlotsTable.tsx:166]: add audit logs

File: src/app/admin/packingSlotsTable/PackingSlotsTable.tsx [hunk 6]
```diff
@@ -181,34 +234,58 @@ const PackingSlotsTable: React.FC = () => {
...
+            const baseAuditLog = getBaseAuditLogForPackingSlotAction(
+                "move a packing slot down",
+                row
+            );
+
+            if (swapRowsError) {
+                setErrorMessage(
+                    `Failed to move packing slot (${row.name}) down. Log ID: ${swapRowsError.logId}`
+                );
+                void sendAuditLog({ ...baseAuditLog, wasSuccess: false });
```
Comment (minalui) [inline src/app/admin/packingSlotsTable/PackingSlotsTable.tsx:283]: Should we add logId to this audit log?

Comment (rickaw) [reply src/app/admin/packingSlotsTable/PackingSlotsTable.tsx]: Good point, changed!

**Review by AmeliaIsCoding — APPROVED** [review]
LGTM, although not sure I'm qualified to review cypress code as I haven't done much of it
Good to see the errors being returned instead of thrown, very satisfying

**Pull request merged**